airflow/www/views.py

```diff
@@ -1583,13 +1588,10 @@ class GraphForm(DateTimeWithNumRunsWithDagRunsForm):
             state_token=state_token(dt_nr_dr_data['dr_state']),
             doc_md=doc_md,
             arrange=arrange,
-            operators=sorted(
-                list(set([op.__class__ for op in dag.tasks])),
-                key=lambda x: x.__name__
-            ),
+            operators={v['name']:v for v in ops}.values(),
```

Comment:
Taking `values()` of this dict comprehension simply equals to `ops`?

Reply:
not completely, this is a list of dicts and I want only the unique dict names. Not doing this results in duplicates in the legend.

Reply:
Now I understand, the code is a little confusing though.

You're using the variable `ops` only as an intermediate for storing the list of dicts of Operator attributes (name, ui_color & ui_fgcolor), only to use `{v['name']:v for v in ops}.values()` later on. IMO this in unnecessary complex and makes it hard to read/understand, how about creating the list of unique operators beforehand (with an appropriate name) and passing that to `self.render()`:

```python
unique_ops = {
    op.name: {"name": op.name, "ui_color": op.ui_color, "ui_fgcolor": op.ui_fgcolor}
    for op in task_instances
}.values()

return self.render(
    .....
    operators=unique_ops,
)
```